torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1210,10 +1210,16 @@ void ProcessGroupNCCL::heartbeatMonitor() {
   // we throw exception and make the whole process to be killed.
   const auto exitMsg = c10::str(
       logPrefix(),
-      "NCCL monitor thread timeout. Basically, this could ",
-      "be due to CUDA or NCCL calls being unexpectedly blocking, ",
-      "especially when your program enters a deadlock state in watchdog "
-      "or destructors. If you see this error, please file a bug to PyTorch.");
+      "ProcessGroupNCCL's watchdog got stuck for ",
+      heartbeatTimeoutInSec_,
+      "seconds without making progress in monitoring enqueued collectives. ",
+      "This typically indicates a nccl/cuda API hang blocking the watchdog, ",
+      "and could be triggered by another thread holding the GIL inside a ",
+      "cuda api, or other deadlock-prone behaviors.",
+      "If you suspect the watchdog is not actually stuck and a longer timeout would help, ",
+      "you can either increase the timeout(TORCH_NCCL_HEARTBEAT_TIMEOUT_SEC) to a larger value "
+      "or disable the heartbeat monitor(TORCH_NCCL_ENABLE_MONITORING=0)"
+      "If neither does not help, please file a bug to PyTorch.");
```

Comment:
> If neither does not help, please file a bug

grammar isn't right (double negatie) - "if neither helps" is clearer 

but also i think the meaning is backwards- if increasing the timeout _DOES_ help, file a bug (bc our timeout value is probably wrong).  But if increasing timeout does _NOT_ help, watchdog is fine.

Reply:
Opps, it should be "If neither helps".

If increasing TIMEOUT helps, they can but not need to file an issue because maybe for that case, the timeout is too short? And if it does not help, then it's a hang case, they need to report hang anyway?

Reply:
OK, I updated this part to make it more clearer.